scanpy/experimental/pp/_highly_variable_genes.py

```diff
@@ -184,7 +231,11 @@ def _highly_variable_pearson_residuals(
         if clip < 0:
             raise ValueError("Pearson residuals require `clip>=0` or `clip=None`.")
 
-        if sp_sparse.issparse(X_batch):
+        if isinstance(X_batch, DaskArray):
```

Comment:
having _calculate_res_dense and _calculate_res_sparse was there before, maybe singledispatch cleaner here.

the non-jitted dask computation is ~10x slower at the moment.